charts/agent-control-deployment/templates/_helpers.tpl

```diff
@@ -125,10 +127,24 @@ cluster name, licenses, and custom attributes
 
 {{- /* Overwrite $config with everything in `config.agentControl.content` if present */ -}}
 {{- $config = mustMergeOverwrite $config (deepCopy (((.Values.config).agentControl).content | default dict)) -}}
-{{- $config | toYaml -}}
+
+{{- /* Perform configuration validations */ -}}
+{{- if not $config.server.enabled -}}
+  {{- fail "The status server cannot be disabled as it is used in the Agent Control container probes" -}}
+{{- end -}}
+{{- if ne $config.server.host $statusServerHost -}}
+  {{- fail "The status server needs to listen on 0.0.0.0 to be used in container probes" -}}
+{{- end -}}
+{{- if ne (printf "%v" $config.server.port) (printf "%v" $statusServerPort) -}}
+  {{- fail "Setting up the status server port is `.Values.config.agentControl.content` is not supported because it would conflict with container probes. Use `.Values.config.status_server.port` instead" -}}
 {{- end -}}
 
+{{- $config | toYaml -}}
+{{- end -}}
 
+{{- define "newrelic-agent-control.config.statusServer.port" -}}
+{{ ((.Values.config).status_server).port | default 51200 }}
+{{- end -}}
```

Comment:
I am a bit confused, the default is already in the values.yaml... why do we need this function?

Reply:
It should update the defaults values.yaml. 

You can try manually to do the upgrade, but I believe it should work. Maybe it is `ct` misbehaving